pandas/core/dtypes/common.py

```diff
@@ -1986,6 +1986,9 @@ def pandas_dtype(dtype):
     # registered extension types
     result = _pandas_registry.find(dtype) or registry.find(dtype)
     if result is not None:
+        # ensure result is an instantiated type
+        if isinstance(result, type):
```

Comment:
can you explain where exactly you think this needed? this violates the return values of this function and should not be needed here. This function returns a type not an instance

Reply:
ok great!

Reply:
One place that this did come up is in `is_bool_dtype` though. I think there a bug there. I can open an issue about it if you want, but basically what happens is if an a EA class is passed in it will return a property method. This can create unexpected side effects. for example:

```python
if is_bool_dtype(Int64dtype):
    print('is a bool dtype')
else:
    print('not a bool dtype')
```

This will print `is a bool dtype` because `is_bool_dtype` won't return `True' or `False`, it will actually return a `property_method` reference. 

That is (one of) the reasons I thought that I needed this.